src/azul/service/async_manifest_service.py

```diff
@@ -63,40 +78,72 @@ class AsyncManifestService(AbstractService):
+                        raise StateMachineError('Maximum manifest generation retries reached.',
+                                                token.execution_id)
+                    output, status = self._inspect_generation(token)
+                    if status not in Status:
+                        raise StateMachineError(status, output)
+                    if status is Status.running:
+                        return token.advance(wait_time=self._get_next_wait_time(token.request_index))
+                    else:
+                        # If a manifest expires before the name of the Step
+                        # Function execution that created it can be reused
+                        # (90 days), StartExecution will fail with
+                        # ExecutionAlreadyExists. In that case we need to retry
+                        # even a successful execution, so that it can recreate
+                        # the manifest.
+                        continue
+                else:
+                    raise
+            else:
+                return token
 
     def inspect_generation(self, token) -> Union[Token, JSON]:
-        try:
-            execution = self.step_function_helper.describe_execution(state_machine_name=self.state_machine_name,
-                                                                     execution_name=token.execution_id)
-        except ClientError as e:
-            if e.response['Error']['Code'] == 'ExecutionDoesNotExist':
-                raise InvalidTokenError from e
-            else:
-                raise
-        output = execution.get('output', None)
-        status = execution['status']
-        if status == 'SUCCEEDED':
+        output, status = self._inspect_generation(token)
+        if status is Status.succeeded:
             # Because describe_execution is eventually consistent, output may
             # not yet be present
             if output is None:
                 return token.advance(wait_time=1)
             else:
                 return json.loads(output)
-        elif status == 'RUNNING':
+        elif status is Status.running:
             return token.advance(wait_time=self._get_next_wait_time(token.request_index))
         else:
-            raise StateMachineError(status, output)
+            raise StateMachineError(status.value, output)
+
+    def _inspect_generation(self, token: Token) -> Tuple[str, Status]:
+        try:
+            helper = self.step_function_helper
+            name = self.state_machine_name
+            execution = helper.describe_execution(state_machine_name=name,
+                                                  execution_name=token.execution_id)
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'ExecutionDoesNotExist':
+                raise InvalidTokenError from e
+            else:
+                raise
+        output = execution.get('output')
+        status = getattr(Status, execution['status'].lower())
```

Comment:
This call to lower() here is brittlely coupled with the one in the enum constructor. The relationship between the enumeration item name and its value should be encapsulated in the enumeration class.